examples/scripts/lora_example.py

```
@@ -0,0 +1,370 @@
+    # Validate quantization options
+    if args.qlora and args.load_in_8bit:
+        print("❌ Error: --qlora (4-bit) and --load-in-8bit cannot be used together. Please choose one.")
+        sys.exit(1)
+
+    # Check for multi-GPU setup
+    is_distributed = 'LOCAL_RANK' in os.environ or 'WORLD_SIZE' in os.environ
+
+    # Warn if user wants multi-GPU but didn't launch with torchrun
+    if args.nproc_per_node > 1 and not is_distributed:
+        print("⚠️  WARNING: You specified --nproc-per-node > 1 but aren't using torchrun!")
+        print(f"   For multi-GPU training, use:")
```

Ruff (0.14.5)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)